elyra/pipeline/kfp/kfp_properties.py

```diff
@@ -0,0 +1,158 @@
+from elyra.pipeline.properties import PipelineParameter
+from elyra.pipeline.properties import PropertyInputType
+
+
+class KfpPropertyInputType(PropertyInputType):
+    """
+    An object representing a single allowed input type for a PropertyAttribute
+    object for KUBEFLOW_PIPELINES runtime processors.
+    """
+
+    _kfp_defined_types = {
+        "String": {"type_hint": "str", "json_type": "string", "default_value": ""},
+        "Bool": {"type_hint": "bool", "json_type": "boolean", "default_value": False, "placeholder": " "},
+        "Integer": {"type_hint": "int", "json_type": "integer"},
+        "Float": {"type_hint": "float", "json_type": "number", "render_input_value": True},
+        # "GCSPath": {"type_hint": "GCSPath", "json_type": "string", "default_value": "", "placeholder": "gs://"},
+        # "GCRPath": {"type_hint": "GCRPath", "json_type": "string", "default_value": "", "placeholder": "gcr.io/"},
+        # "GCPRegion": {"type_hint": "GCPRegion", "json_type": "string", "default_value": ""},
+        # "GCPProjectID": {"type_hint": "GCPProjectID", "json_type": "string", "default_value": ""},
+        "CustomString": {"json_type": "string", "type_title": "String-valued parameter of arbitrary type"}
+        # "List": {"type_hint": "list", "json_type": "array", "default_value": []},  # not yet supported by frontend
+        # "Dict": {"type_hint": "dict", "json_type": "object", "default_value": {}},  # not yet supported by frontend
+    }
+
+    def __init__(
+        self,
+        base_type: str,
+        default_value: Optional[Any] = None,
+        placeholder: Optional[Any] = None,
+        enum: Optional[List[Any]] = None,
+        **kwargs,
+    ):
+        super().__init__(
+            base_type=base_type,
+            default_value=default_value,
+            placeholder=placeholder,
+            enum=enum,
+            runtime_defined_types=self._kfp_defined_types,
+            **kwargs,
+        )
+
+        self.type_hint = kwargs.get("type_hint") or self._kfp_defined_types[base_type].get("type_hint")
+        self.component_input_type = kwargs.get("type_hint") or self.base_type
+
+
+class KfpPipelineParameter(PipelineParameter):
+    """An ElyraProperty representing a single pipeline parameter for the Kubeflow Pipelines runtime"""
+
+    property_id = "KFP_PIPELINE_PARAMETERS"
+    property_attributes = [
+        ListItemPropertyAttribute(
+            attribute_id="name",
+            description="The name of the parameter (must be a valid python variable name)",
+            display_name="Parameter Name",
+            allowed_input_types=[PropertyInputType(base_type="str", placeholder="param_1")],
+            hidden=False,
+            required=True,
+            use_in_key=True,
+            pattern="^[a-zA-Z][a-zA-Z0-9_]*$",
+        ),
+        ListItemPropertyAttribute(
+            attribute_id="description",
+            description="A description for this parameter",
```

Comment:
Not. to be consistent with help text format used for properties.

```suggestion
            description="A description for this parameter.",
```